### ISSUE TITLE
Is this project still alive

### ISSUE THREAD

**Issue by taste66** [gkaindl/Community-mdnsResponder#1]
I am interested in the state of this project. Is it still active and in use?

**Comment by victorgama** [comment]
@gkaindl Are you still maintaining this project?

**Comment by gkaindl** [comment]
actually, I haven't done any development on it lately, but I'm still using this mDNSResponder on a lot of embedded linux devices together with my wide-area bonjour domain hosted on AWS (e.g. dnsextd).

since Apple has switched to discoveryd in 10.10, which is closed-source, I plan to port my patches to the last version of the open-source mDNSResponder package they released, but I do not have an ETA.

however, as long as you do not need IPv6 service registration on wide-area bonjour, this version should just work fine as it is right now!

**Comment by victorgama** [comment]
Perfect! I just mentioned on a mdns node.js module that they should ditch avahi and use mdnsResponder from this repository. I am using it without problems. Thank you for porting the code. :beer:

**Comment by victorgama** [comment]
@gkaindl Just a heads up: looks like Apple will ditch discoveryd in future versions of OS X.

http://daringfireball.net/linked/2015/05/26/discoveryd

**Comment by gkaindl** [comment]
@victorgama haha, yes, I was quite delighted to see this, not so much because of this project, but because I'm amongst those continually bitten by the fact that discoveryd just. dies. all. the. time! :-D

**Comment by victorgama** [comment]
@gkaindl Exactly! I struggle with the very same problem, but on my case, when it don't just die, I have to kill it. :-1:

**Comment by choikwa** [comment]
When I tried to build it, I get yyparse((void*)d) issue [yyparse(void), no params].
I also get 
`
../mDNSShared/dnsextd_parser.y:189:22:` error: ‘context’ undeclared (first use in this function)
    ( ( DaemonInfo* ) context )->private_port = mDNSOpaque16fromIntVal( $3 );
                      ^
`

**Comment by thor-the-norseman** [comment]
I'm looking at this for a different use case: Discovery of microservices.

Thought process: Daemons exist for this, but then I'm left discovering _that_ via DNS. What if the daemon is down and I'd like to let my clients know about the new one I just launched on this other VPS? Then I'd need to update my DNS records. I'd hate to do that manually, so is there a way to do it automatically? Ah, BIND9 supports secure dynamic DNS updates. Hang on, I'm basically doing service discovery via DNS. Could I just drop the discovery daemon and use DNS instead? That would be much better, since the entire DNS ecosystem would provide redundancy. Except, how would I auto-erase old records? If a service dies, it can't erase them. Oh, so that's what Bonjour is for. Great! Let's use dnsextd. What? It hasn't been ported to Linux? There must be a port. There has to be a port! Yay! There is a port. Aww, the maintainer hasn't updated his repo for several years. :(

And now I'm here and I urgently need service discovery via DNS for my project.

I realize that record changes to my domain would take time to propagate up the chain, but this is still the best solution I have found for service discovery. DNS was designed for redundancy from the ground up. My service records would be visible anywhere on the Internet without a single point of failure.

I don't necessarily need to use SRV records for this; they'd be hard to look up in a browser, for example, but that could be solved by using A records instead and standardizing on the port numbers. As far as I can tell, dnsextd doesn't really care if you conform to Bonjour, and simply adds support for a couple of IETF drafts to BIND9.

**Comment by stokito** [comment]
FYI: For OpenWrt there is now a dedicated https://openwrt.org/docs/guide-developer/mdns
Hope it will be useful
